sql/moz-fx-data-shared-prod/adjust_derived/adjust_cohort_v1/query.py

```diff
@@ -0,0 +1,254 @@
+"""Adjust data - download deliverables, clean and upload to BigQuery."""
+import csv
+import json
+import os
+import tempfile
+from argparse import ArgumentParser
+
+import requests
+from google.cloud import bigquery
+
+API_URI = "api.adjust.com"
+ENDPOINT = "kpis"
+API_VERSION = "v1"
+
+APP_TOKEN_LIST = os.environ.get("ADJUST_APP_TOKEN_LIST")
+
+"""This is Marlene's personal Adjust API Token. We don't have a special service account API Token."""
+API_TOKEN = os.environ.get("ADJUST_API_TOKEN")
+
+"""The APP_TOKEN_LIST is a list of dictionaries.
+Keys are the app_name, and app_token in the form:
+{"app_name":"<specific app name from Adjust dash>". "app_token":"<unique Adjust app token>}"
+Look here to see the apps we are tracking: https://dash.adjust.com/#/.
+It is important to maintain this list in order for the script to work especially in the case of new apps
+being added to track in Adjust
+"""
+
+CSV_FIELDS = [
+    "date",
+    "period_length",
+    "period",
+    "app",
+    "app_token",
+    "network",
+    "network_token",
+    "country",
+    "os",
+    "device",
+    "cohort_size",
+    "retained_users",
+    "retention_rate",
+    "time_spent_per_user",
+    "time_spent_per_session",
+    "time_spent",
+    "sessions_per_user",
+    "sessions",
+]
+
+
+def read_json(filename: str) -> dict:
+    """Read JSON file."""
+    with open(filename, "r") as f:
+        data = json.loads(f.read())
+    return data
+
+
+def write_dict_to_csv(json_data, filename):
+    """Write a dictionary to a csv."""
+    with open(filename, "w") as out_file:
+        dict_writer = csv.DictWriter(out_file, CSV_FIELDS)
+        dict_writer.writeheader()
+        dict_writer.writerows(json_data)
+
+
+def download_adjust_kpi_data(date, api_token, app_token):
+    """Download data from Adjust - API token and APP tokens are called here."""
+    start_date = date
+    end_date = date
+
+    kpis = [
+        "cohort_size",
+        "retained_users",
+        "retention_rate",
+        "time_spent_per_user",
+        "time_spent_per_session",
+        "time_spent",
+        "sessions_per_user",
+        "sessions",
+    ]
+    groupings = [
+        "days",
+        "periods",  # from result_parameter
+        "apps",
+        "networks",
+        "countries",
+        "os_names",
+        "device_types",
+    ]
+    # getting overview metrics for different kpis / Deliverables
+    url = f"https://api.adjust.com/kpis/v1/{app_token}/cohorts"  # overview
+    url_params = f"start_date={start_date}&end_date={end_date}&kpis={','.join(kpis)}&grouping={','.join(groupings)}"
+    headers = {
+        "Authorization": f"Bearer {api_token}",
+    }
+
+    response = requests.get(url, headers=headers, params=url_params)
```

Comment:
Check the value of response.status_code != 200 and log an error.  Specifically check for 401 (Auth)  and Bad Request (400)